### PR TITLE
FINERACT-2082: Progressive loan schedule type should only support Adv…

### PR DESCRIPTION
**Pull request by alberto-art3ch**
…anced payment allocation strategy

## Description

We have added a validation to avoid create and edit Loan Product with Progressive Loan Schedule type and other repayment strategy different than Advanced Payment Allocation

[FINERACT-2082](https://issues.apache.org/jira/browse/FINERACT-2082)

Screenshot 2024-05-16 at 2 35 33 p m

## Checklist

Please make sure these boxes are checked before submitting your pull request - thanks!

- [ ] Write the commit message as per https://github.com/apache/fineract/#pull-requests

- [ ] Acknowledge that we will not review PRs that are not passing the build _("green")_ - it is your responsibility to get a proposed PR to pass the build, not primarily the project's maintainers.

- [ ] Create/update unit or integration tests for verifying the changes made.

- [ ] Follow coding conventions at https://cwiki.apache.org/confluence/display/FINERACT/Coding+Conventions.

- [ ] Add required Swagger annotation and update API documentation at fineract-provider/src/main/resources/static/legacy-docs/apiLive.htm with details of any API changes

- [ ] Submission is not a "code dump".  (Large changes can be made "in repository" via a branch.  Ask on the developer mailing list for guidance, if required.)

FYI our guidelines for code reviews are at https://cwiki.apache.org/confluence/display/FINERACT/Code+Review+Guide.

### CODE REVIEW

File: fineract-provider/src/main/java/org/apache/fineract/portfolio/loanproduct/serialization/LoanProductDataValidator.java [hunk 1]
```diff
@@ -809,6 +809,14 @@ public void validateForCreate(final JsonCommand command) {
                     "Progressive repayment schedule processing is only available with `Advanced payment allocation` strategy");
         }
 
+        if (LoanScheduleType.PROGRESSIVE.equals(LoanScheduleType.valueOf(loanScheduleType))
```
Comment (adamsaghy) [inline fineract-provider/src/main/java/org/apache/fineract/portfolio/loanproduct/serialization/LoanProductDataValidator.java]: Please extract this validation into a method and that way can be reused when loan got submitted or when loan got modified!

**Review by adamsaghy — CHANGES_REQUESTED** [review]
Kindly see my recommendation!

File: fineract-provider/src/main/java/org/apache/fineract/portfolio/loanproduct/serialization/LoanProductDataValidator.java (hunk 1 above)
Comment (alberto-art3ch) [reply fineract-provider/src/main/java/org/apache/fineract/portfolio/loanproduct/serialization/LoanProductDataValidator.java]: Done! code updated!

**Review by adamsaghy — APPROVED** [review]
LGTM

**Pull request merged**